src/Lean/Data/Lsp/Extra.lean

```diff
@@ -8,20 +8,22 @@ import Lean.Data.Json
 import Lean.Data.JsonRpc
 import Lean.Data.Lsp.Basic
 
-/-
+/-!
 This file contains Lean-specific extensions to LSP.
 The following additional packets are supported:
-- "textDocument/waitForDiagnostics": Yields a response when all the diagnostics that were pending at the time of
-  arrival of the packet have been emitted. Exists for synchronization purposes, e.g. during testing or when external tools might want to use
-  our LSP server.
+- "textDocument/waitForDiagnostics": Yields a response when all the diagnostics for a version of the document
```

Comment:
The edit delay has the side effect of reordering messages. For example, a request sent by the client immediately after a `didChange` notification will get to the worker *before* said change. I *think* this is okay w.r.t. all the functionality we support right now (e.g. it's ok to briefly reply with slightly outdated hovers). But this breaks anything expected to be deterministic like `waitForDiagnostics`. To fix this, it now includes a version specifier. And this will be something we'll have to keep in mind for future functionality. It should also be possible to block all messages until the edit delay expires, but that seems convoluted and currently unnecessary.